vertical-pod-autoscaler/pkg/recommender/model/cluster.go

```diff
@@ -292,13 +292,23 @@ func (cluster *ClusterState) findOrCreateAggregateContainerState(containerID Con
 }
 
 // GarbageCollectAggregateCollectionStates removes obsolete AggregateCollectionStates from the ClusterState.
+// AggregateCollectionState is obsolete in two situations:
+// 1) It has no samples and there are no more active pods that can contribute,
+// 2) The last sample is too old to give meaningful recommendation.
 func (cluster *ClusterState) GarbageCollectAggregateCollectionStates(now time.Time) {
 	klog.V(1).Info("Garbage collection of AggregateCollectionStates triggered")
 	keysToDelete := make([]AggregateStateKey, 0)
+	activeKeys := cluster.getActiveAggregateStateKeys()
 	for key, aggregateContainerState := range cluster.aggregateStateMap {
+		isKeyActive := activeKeys[key]
```

Comment:
What do you think about making activeKeys map[AggregateStateKey]struct{} and checking here if the key is present (instead of checking if it maps to true)?

Reply:
It's basically the same (both are correct ways of representing a set in go) and I really hate writing activeKeys[cluster.MakeAggregateStateKey(pod, container)] = struct{}{}